pycbc/inference/models/gaussian_noise.py

```diff
@@ -852,7 +856,59 @@ def _nowaveform_loglr(self):
             setattr(self._current_stats, '{}_optimal_snrsq'.format(det), 0.)
         return -numpy.inf
 
-    def _loglr(self):
+    def update(self, **params):
+        # update
+        super().update(**params)
+        # reset current waveforms
+        self._current_wfs = None
+
+    @property
+    def multi_signal_support(self):
+        """ The list of classes that this model supports in a multi-signal
+        likelihood
+        """
+        return [type(self)]
+
+    def multi_loglikelihood(self, models):
+        """ Calculate a multi-model (signal) likelihood
+        """
+        # Generate the waveforms for each submodel
+        wfs = []
+        for m in models:
+            wfs.append(m.get_waveforms())
```

Comment:
It's odd that you're not calling `self`'s `get_waveforms` here. You're relying on the user passing in `self` in the list of models, which isn't evident from the API. Naively, I would think that calling `model.multi_loglikelihood(models)` would include that model in the loglikelihood, otherwise, why would this be an instance method and not a stand-alone function?